### PR TITLE
aws credentials: fix task role expiration parsing

### PR DESCRIPTION
**Pull request by nbaws**
Commit Message: aws credentials: fix task role expiration parsing , aws credentials: fix test cases with correct date/time format
Additional Description: Bug fix, task role expiration does not appear to have ever been parsed correctly. This causes the task role metadata to be hit 100% of cases, rather than once per hour.
Risk Level: Low
Testing: Unit
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by nbaws** [comment]
An example of task role credentials, with expiration in RFC3339 format. This patch amends expiration pattern match to this format.
`{"RoleArn":"arn:aws:iam::123456:role/envoy-task-role","AccessKeyId":"ASIAQXXXXXXOI","SecretAccessKey":"*","Token":"*","Expiration":"2023-08-26T10:49:38Z"}`

File: test/extensions/common/aws/credentials_provider_impl_test.cc [hunk 1]
```diff
@@ -646,7 +646,7 @@ TEST_F(TaskRoleCredentialsProviderTest, TimestampCredentialExpiration) {
   "AccessKeyId": "new_akid",
   "SecretAccessKey": "new_secret",
   "Token": "new_token",
```
Comment (suniltheta) [inline test/extensions/common/aws/credentials_provider_impl_test.cc:648]: Would you be able to include this additional unit test?
[pr_29267_test.txt](https://github.com/envoyproxy/envoy/files/12446644/pr_29267_test.txt)

**Review by suniltheta — CHANGES_REQUESTED** [review]
An additional test that refresh if the credentials expire period is less than 1hr refresh interval

File: test/extensions/common/aws/credentials_provider_impl_test.cc (hunk 1 above)
Comment (suniltheta) [reply test/extensions/common/aws/credentials_provider_impl_test.cc:648]: Nevermind, I think we should be good with `TimestampCredentialExpiration` test.

**Review by suniltheta — APPROVED** [review]

**Comment by nbaws** [comment]
@mattklein123 would you be able to take a look at this one - we are hoping to pull it back to a local fork once merged to upstream.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**